### ISSUE TITLE
A multi-statement transaction can not use multiple storage engines (sequence + vinyl)

### ISSUE THREAD

**Issue by rk108** [tarantool/tarantool#4624]
Tarantool version:
Tarantool 1.10.4-12-g6f8ce4b0d
Target: Linux-x86_64-RelWithDebInfo
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_BACKTRACE=ON
Compiler: /usr/bin/cc /usr/lib/ccache/g++
C_FLAGS:-g -O2 -fdebug-prefix-map=/build/tarantool-1.10.4.12=. -specs=/usr/share/dpkg/no-pie-compile.specs -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -fno-gnu89-inline -Wno-cast-function-type
CXX_FLAGS:-g -O2 -fdebug-prefix-map=/build/tarantool-1.10.4.12=. -specs=/usr/share/dpkg/no-pie-compile.specs -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-cast-function-type

OS version:
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"

Bug description:
error: A multi-statement transaction can not use multiple storage engines
with box.sequence.seq_name:next()
in box.begin() box.space.space_name:replace({....})
where space_name is vinyl engine

Steps to reproduce:

box.schema.sequence.create('super_seq')
local s = box.schema.space.create('seq_vinyl', {engine = 'vinyl'})

s:create_index('primary', {type = 'tree',
                parts = {1, 'string'},
                unique = true})

box.begin()

box.sequence.super_seq:next()
box.space.seq_vinyl:replace({'test', 0})

- error: A multi-statement transaction can not use multiple storage engines

**Comment by Korablev77** [comment]
I guess it is blocked by https://github.com/tarantool/tarantool/issues/1958

**Comment by kyukhin** [comment]
I guess, this is duplicate of #1958
